app/views/recommended_links/_update_form.html.erb

```diff
@@ -0,0 +1,23 @@
+<%= form_for(recommended_link) do |f| %>
```

Comment:
I just had another look and I just realised why this looks weird to me.
You are not following rails conventions.
Normally this is done with a `new.html.erb` and `edit.html.erb` both then use `_form.html.erb` partial which deals with the _special_ cases, which in this case is to omit `keywords` and `search_index` from the edit form